src/main/scala/org/ergoplatform/ErgoLikeTransaction.scala

```diff
@@ -69,20 +67,19 @@ class ErgoLikeTransaction(override val inputs: IndexedSeq[Input],
 
   require(outputCandidates.length <= Short.MaxValue, s"${Short.MaxValue} is the maximum number of outputs")
 
-  override type IdType = Digest32
-
-  override lazy val id: IdType = Blake2b256.hash(messageToSign)
+  override lazy val id: ModifierId = Blake2b256.hash(messageToSign).toModifierId
 
   override def equals(obj: Any): Boolean = obj match {
-    case tx: ErgoLikeTransaction => this.id sameElements tx.id  //we're ignoring spending proofs here
+    //we're ignoring spending proofs here
+    case tx: ErgoLikeTransaction => this.id == tx.id
```

Comment:
@greenhat if we override equals we should also override hashCode. I understand that is was like this, but please fix it.

Reply:
@ergomorphic I should've spotted this myself. Good catch! Thanks!